app/models/manageiq/providers/vmware/network_manager/refresh_parser.rb

```diff
@@ -28,36 +28,70 @@ def ems_inv_to_hashes
 
     private
 
-    def get_networks
-      # fetch org VDC networks
-      @inv[:networks] = @org.networks
+    def get_vdc_networks
+      @inv[:vdc_networks] = @org.networks
+      @inv[:vdc_networks_idx] = Hash[@inv[:vdc_networks].map { |net| [net.id, net] }]
 
-      # fetch vApp networks
-      @inv[:networks] += get_vapp_networks
+      process_collection(@inv[:vdc_networks], :cloud_networks) { |n| parse_network(n) }
+      process_collection(@inv[:vdc_networks], :cloud_subnets, false) { |n| parse_network_subnet(n) }
 
-      process_collection(@inv[:networks], :cloud_networks) { |n| parse_network(n) }
-      process_collection(@inv[:networks], :cloud_subnets, false) { |n| parse_network_subnet(n) }
+      $vcloud_log.info("#{log_header} Fetched #{@inv[:vdc_networks].count} VDC networks")
     end
 
-    def get_network_ports
-      # TODO: Update FOG so that it will provide list of network ports. Currently, it is only possible to obtain
-      # very little data, barely enough to tell to which network the VM is connected to.
-      @inv[:vm_networks] = []
-      @org.vdcs.each do |vdc|
-        vdc.vapps.each do |vapp|
-          vapp.vms.each do |vm|
-            @inv[:vm_networks] += vm.network_adapters.each_with_index.map do |net_if, i|
-              net_if[:vapp] = vapp
-              net_if[:vm] = vm
-              net_if[:vm_network] = vm.network
-              net_if[:idx] = i
-              net_if
-            end
+    def get_vapp_networks
+      @inv[:vapp_networks] = []
+      @inv[:routers] = []
+
+      @ems.orchestration_stacks.each do |stack|
+        fetch_network_configurations_for_vapp(stack.ems_ref).map do |net_conf|
+          $vcloud_log.debug("#{log_header} processing net_conf for vapp #{stack.ems_ref}: #{net_conf}")
+          network_id = network_id_from_links(net_conf)
+          $vcloud_log.debug("#{log_header} calculated vApp network id: #{network_id}")
+          if (vdc_net = corresponding_vdc_network(net_conf, @inv[:vdc_networks_idx]))
+            $vcloud_log.debug("#{log_header} skipping VDC network duplicate")
+            memorize_network_name_mapping(stack.ems_ref, vdc_net.name, vdc_net.id)
+            next
```

Comment:
Instead of skipping the loop can you just move [this](https://github.com/ManageIQ/manageiq-providers-vmware/pull/161/files#diff-b90e0cdaee1f8e3c7c23889a3fc00f93R60) into the `else`?

Reply:
👍 shouldn't need the next now